corehq/apps/app_manager/models.py

```diff
@@ -1921,12 +1922,18 @@ def from_json(cls, data):
             for item in to_ret.enum:
                 for lang, path in item.value.items():
                     item.value[lang] = interpolate_media_path(path)
+
+        if to_ret.optimization and not to_ret.supports_optimizations:
+            to_ret.optimization = None
```

Comment:
just a stronger check on the model to read optimization from data from view only when supported.

Reply:
Definitely not necessary.

It prevents any misconfiguration sent from the UI, if any. Just model validation to not rely on UI checks only.

>  what would break if the suite includes optimizations but they're not supported on the CC version?

So, this is a check for the format of the column. Optimization is supported only for specific ones.

The suite file has a check as well to only add optimizations when supported for the column.
In case it's incorrectly set for a format that does not support it, it might break the app or cause unexpected app behavior, because CC would try to optimize it, if the CC version is up to date.